app/src/androidTest/java/com/popularMovies/tests/Helpers.java

```diff
@@ -0,0 +1,204 @@
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.withDecorView;
+import static android.support.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Created by ioana.hoaghia on 2/14/2018.
+ */
+
+public class Helpers extends EspressoTestBase {
+
+    public static int getRecyclerViewChildCount(final Matcher<View> matcher) {
+        final int[] count = {0};
+        onView(matcher).perform(new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return isAssignableFrom(android.support.v7.widget.RecyclerView.class);
+            }
+
+            @Override
+            public String getDescription() {
+                return "getting child count";
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                android.support.v7.widget.RecyclerView rv = (android.support.v7.widget.RecyclerView) view;
+                count[0] = rv.getChildCount();
+            }
+        });
+        return count[0];
+    }
+
+
+    public static int getGridViewChildCount(final Matcher<View> matcher) {
+        final int[] count = {0};
+        onView(matcher).perform(new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return isAssignableFrom(android.widget.GridView.class);
+            }
+
+            @Override
+            public String getDescription() {
+                return "getting child count";
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                android.widget.GridView gv = (android.widget.GridView) view;
+                count[0] = gv.getChildCount();
+            }
+        });
+        return count[0];
+    }
+
+    public static Matcher<View> childAtPosition(final Matcher<View> parentMatcher, final int position) {
+
+        return new TypeSafeMatcher<View>() {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("Child at position " + position + " in parent ");
+                parentMatcher.describeTo(description);
+            }
+
+            @Override
+            public boolean matchesSafely(View view) {
+                ViewParent parent = view.getParent();
+                return parent instanceof ViewGroup && parentMatcher.matches(parent) && view.equals(((ViewGroup) parent).getChildAt(position));
+            }
+        };
+    }
+
+    public static void clickOnItem (int id)
```

Comment:
Dupa nume, asta pare o metoda mai generala (item = obiectul pe care vrem sa dam click = ViewMatcher). Deci, in loc sa-i dam ca parametru doar ID-ul, ii putem da un Matcher<View> si putem folosi metoda peste tot unde ai facut ".perform(click());". :D  

Daca vrei sa pastrezi metoda in forma actuala, o poti redenumi "clickOnItemWithId".

Reply:
Da, face sens. Am lasat-o si pe clickOnItemWithId, si am adaugat si clickOnItem + matcher.